src/pybind/mgr/volumes/fs/nfs.py

```diff
@@ -0,0 +1,454 @@
+            if self.rados_namespace:
+                ioctx.set_namespace(self.rados_namespace)
+            ioctx.remove_object("export-{}".format(export_id))
+
+    def _save_export(self, export):
+        export.fsal.cephx_key = self.cephx_key
+        self.exports[export.export_id] = export
+        conf_block = export.to_export_block()
+        self._write_raw_config(conf_block, "export-{}".format(export.export_id))
+        self._persist_daemon_configuration()
+
+    def create_export(self, ex_dict):
+        ex_id = self._gen_export_id()
+        export = Export.from_dict(ex_id, ex_dict)
+        self._save_export(export)
+        return ex_id
+
+    def remove_export(self, export_id):
+        if export_id not in self.exports:
+            return None
+        export = self.exports[export_id]
+        del self.exports[export_id]
+        self._delete_export(export_id)
+        return export
+
+    def has_export(self, export_id):
+        return export_id in self.exports
+
+    def list_daemons(self):
+        return [daemon_id for daemon_id in self.daemons_conf_blocks]
+
+    def reload_daemons(self, daemons):
+        with self.mgr.rados.open_ioctx(self.rados_pool) as ioctx:
+            if self.rados_namespace:
+                ioctx.set_namespace(self.rados_namespace)
+            for daemon_id in daemons:
+                ioctx.notify("conf-{}".format(daemon_id))
+
+    def make_rados_url(self, obj):
+        if self.rados_namespace:
+            return "rados://{}/{}/{}".format(self.rados_pool, self.rados_namespace, obj)
+        return "rados://{}/{}".format(self.rados_pool, obj)
+
+class NFSConfig(object):
+    exp_num = 0
+
+    def __init__(self, mgr, cluster_id):
+        self.cluster_id = "ganesha-%s" % cluster_id
+        self.pool_name = 'nfs-ganesha'
+        self.pool_ns = cluster_id
+        self.mgr = mgr
+        self.ganeshaconf = None # type: Optional[GaneshaConf]
+        self.key = ''
+
+    def update_user_caps(self):
+        if NFSConfig.exp_num > 0:
+            ret, out, err = self.mgr.mon_command({
+                'prefix': 'auth caps',
+                'entity': "client.%s" % (self.cluster_id),
+                'caps' : ['mon', 'allow *', 'osd', 'allow * pool=%s namespace=%s, allow rw tag cephfs data=a' % (self.pool_name, self.pool_ns), 'mds', 'allow * path=/'],
+                })
+
+            if ret!= 0:
+                return ret, out, err
+
+    def create_common_config(self, nodeid):
+        # TODO change rados url to "%url rados://{}/{}/{}".format(self.pool_name, self.pool_ns, nodeid)
+        result = """NFS_CORE_PARAM {{
+        Enable_NLM = false;
+        Enable_RQUOTA = false;
+        Protocols = 4;
+        }}
+
+        CACHEINODE {{
+        Dir_Chunk = 0;
+        NParts = 1;
+        Cache_Size = 1;
+        }}
+
+        NFSv4 {{
+        RecoveryBackend = rados_cluster;
+        Minor_Versions = 1, 2;
+        }}
+
+        RADOS_URLS {{
+        userid = {2};
+        }}
```

Comment:
I think you also want a watch_url directive here, so that a RADOS notify on that object will cause a config reload. It's currently borked in v3.2, apparently but should show up in v3.3 (and we may carry a one-off fix for it, eventually).

Reply:
As per the email discussion, watch_url directive has some issues. In this PR I will just focus on creating exports. Here we can do without it for now. Once we find solution for the watch_url issues, I will add it. Is it fine ?

Reply:
As per the email discussion, watch_url directive has some issues. I will add it once we find solution for it.

Reply:
Sounds reasonable to me!